### PR TITLE
[fungibleAssets] Ensure query for coin balances uses correct query

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
For migrated coins, they need to query both the address and the coin type associated.

### CODE REVIEW

File: src/internal/account.ts [hunk 1]
```diff
@@ -404,16 +405,39 @@ export async function getAccountTransactionsCount(args: {
...
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
+  } else if (coinType !== undefined && faMetadataAddress === undefined) {
+    // TODO Move to a separate function as defined in the AIP for coin migration
+    if (args.coinType === APTOS_COIN) {
+      faAddress = AccountAddress.A.toStringLong();
+    } else {
+      faAddress = createObjectAddress(AccountAddress.A, coinType).toStringLong();
+    }
+  } else if (coinType === undefined && faMetadataAddress !== undefined) {
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
```
Comment (rtso) [inline src/internal/account.ts:424]: If faMetadataAddress is set, should this convert to the coinType?

File: src/internal/account.ts [hunk 2]
```diff
@@ -404,16 +405,39 @@ export async function getAccountTransactionsCount(args: {
...
+  let faAddress: string;
+
+  if (coinType !== undefined && faMetadataAddress !== undefined) {
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
+  } else if (coinType !== undefined && faMetadataAddress === undefined) {
+    // TODO Move to a separate function as defined in the AIP for coin migration
+    if (args.coinType === APTOS_COIN) {
+      faAddress = AccountAddress.A.toStringLong();
+    } else {
+      faAddress = createObjectAddress(AccountAddress.A, coinType).toStringLong();
```
Comment (rtso) [inline src/internal/account.ts:421]: @lightmark to check that this function is correct

File: src/internal/account.ts (hunk 1 above)
Comment (gregnazario) [reply src/internal/account.ts:424]: There's no way to retrieve the coin type from the FA metadata address.

So, in this case I'm taking it as an FA by itself.

File: src/internal/account.ts (hunk 2 above)
Comment (gregnazario) [reply src/internal/account.ts:421]: I took this from the code directly here, which should be extracted to a view function:

https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L320-L328

File: src/internal/account.ts (hunk 1 above)
Comment (lightmark) [reply src/internal/account.ts:424]: This is the expected behavior.

File: src/internal/account.ts (hunk 2 above)
Comment (lightmark) [reply src/internal/account.ts:421]: @rtso what is the `asset_type` is not changed from `0x1::aptos_coin::AptosCoin` to `0xa` yet?
From the client perspective, they don't know what to use. Although may be a short term issue.

**Review by lightmark — COMMENTED** [review]
lgtm except the question for @rtso

**Review by lightmark — APPROVED** [review]

**Pull request merged**